src/Symfony/Component/Validator/Constraints/ColorValidator.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class ColorValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Color) {
+            throw new UnexpectedTypeException($constraint, Color::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!\is_string($value) && !(\is_object($value) && method_exists($value, '__toString'))) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        $value = (string) $value;
+
+        if (!preg_match('/^#[0-9a-f]{6}$/i', $value)) {
```

Comment:
See https://www.w3.org/TR/html52/infrastructure.html#valid-simple-color for the spec.

Reply:
The definition of color is really broad, so I don't think it's fair to only validate a hex color this way (ignoring `#FFF` definitions). if this class would be called `HexColorValidator` with a `HexColor` constraint, it would make a lot more sense to me. 

In the current scenarios, I would think "red", "green", "blue" etc would also be valid, or `rgb(255, 0, 0)` for example.